cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringAgentService.java

```diff
@@ -333,8 +334,9 @@ private void publishStartProgram(String runPayload, PeerInfo peerInfo) throws IO
 
   private void publishStopProgram(String stopPayload) {
     ProgramRunInfo programRunInfo = GSON.fromJson(stopPayload, ProgramRunInfo.class);
-    ProgramRunId programRunId = new ProgramRunId(programRunInfo.getNamespace(),
-        programRunInfo.getApplication(),
+    ApplicationId appId = new NamespaceId(programRunInfo.getNamespace())
```

Comment:
I think it would be good to have some common logic for this, since it is very error prone. Maybe a constructor or some utility method in ProgramRunId that takes a ProgramRunInfo and converts it?

Reply:
added ProgramRunId constructor that takes a ProgramRunInfo